src/core_seaice/shared/mpas_seaice_advection_incremental_remap.F

```diff
@@ -6801,214 +6755,166 @@ subroutine integrate_fluxes_over_triangles(&
+                                     + thisTracer %  xGrad3D(iLayer,iCat,iCell) * xTriangle(iqp,iTri,iEdge)  &
+                                     + thisTracer %  yGrad3D(iLayer,iCat,iCell) * yTriangle(iqp,iTri,iEdge)  &
+                                     )
+                               endif
+                               ! integrate over the triangle by summing over quadrature points
+                               tracerIntegral3D(iLayer,iCat) = tracerIntegral3D(iLayer,iCat) &
+                                  + weightQuadPoint(iqp) * thisTracer % triangleValue3D(iLayer,iCat,iqp,iTri,iEdge)
+                            enddo ! iLayer
+                         enddo    ! iCat
+                      enddo       ! iqp
                       ! increment the area-weighted flux across the edge
                       thisTracer % edgeFlux3D(:,:,iEdge) = thisTracer % edgeFlux3D(:,:,iEdge) &
                                                          + triangleArea(iTri,iEdge) * tracerIntegral3D(:,:)
-
                    endif ! triangleArea /= 0
-
                 enddo    ! nTriPerEdgeRemap
-
              endif       ! maskEdge = 1
           enddo          ! nEdges
+          deallocate(tracerIntegral3D)
 
-          ! Check for negative reconstructed ice area
-          if (trim(thisTracer % tracerName) == 'iceAreaCategory') then
-             do iEdge = 1, nEdges
-                do iTri = 1, nTriPerEdgeRemap
-                   if (triangleArea(iTri,iEdge) /= 0.0_RKIND) then
-                      iCell = iCellTriangle(iTri,iEdge)
-                      do iqp = 1, nQuadPoints
-                         do iCat = 1, nCategories
+       endif   ! ndims
+
+       ! Check for negative reconstructed ice area
+       if (trim(thisTracer % tracerName) == 'iceAreaCategory') then
+          do iEdge = 1, nEdges
+             do iTri = 1, nTriPerEdgeRemap
+                if (triangleArea(iTri,iEdge) /= 0.0_RKIND) then
+                   iCell = iCellTriangle(iTri,iEdge)
+                   do iqp = 1, nQuadPoints
+                      do iCat = 1, nCategories
+                         if (thisTracer % ndims == 2) then
+                            if (thisTracer % triangleValue2D(iCat,iqp,iTri,iEdge) < 0.0_RKIND) then
+                               call mpas_log_write('MPAS-seaice: IR negative reconstructed ice area', messageType=MPAS_LOG_ERR)
+                               call mpas_log_write('nCells, iCat, iCell, global iCell: $i $i $i $i',  messageType=MPAS_LOG_ERR, &
+                                    intArgs=(/nCells, iCat, iCell, indexToCellID(iCell)/))
+                               call mpas_log_write('iEdge, global iEdge, iTri, iqp: $i $i $i $i', messageType=MPAS_LOG_ERR, &
+                                    intArgs=(/iEdge, indexToEdgeID(iEdge), iTri, iqp/))
+                               call mpas_log_write('triangle area, tracer, center, xGrad2D, yGrad2D: $r $r $r $r $r', &
```

Comment:
Was there a performance reason the error messages were compressed into a single call? I think its probably more readable as it was before?

Reply:
Yes, it's better to run the normal/error-free case as fast as possible and do error-checking/debug-logging from single-threaded region.